control-admin/src/test/java/fi/nls/oskari/control/admin/LayerAdminUsageCheckHandlerTest.java

```diff
@@ -0,0 +1,83 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({OskariComponentManager.class})
+public class LayerAdminUsageCheckHandlerTest extends JSONActionRouteTest {
+
+    private final static Integer WMS_LAYER_ID = 1;
+    private final static Integer SECOND_WMS_LAYER_ID = 2;
+    private final static Integer THIRD_WMS_LAYER_ID = 3;
+    private final static Integer FOURTH_WMS_LAYER_ID = 4;
+    private final static Integer METADATA_LAYER_ID = 10;
+
+    final private static LayerAdminUsageCheckHandler handler = new LayerAdminUsageCheckHandler();
+
+    @BeforeClass
+    public static void setup() throws JSONException {
+        OskariLayerService mapLayerService = mock(OskariLayerService.class);
+        PowerMockito.mockStatic(OskariComponentManager.class);
+        when(OskariComponentManager.getComponentOfType(OskariLayerService.class)).thenReturn(mapLayerService);
+        doReturn(getTestLayers()).when(mapLayerService).findAll();
+        handler.init();
+    }
+
+    private static List<OskariLayer> getTestLayers() throws JSONException {
+        List<OskariLayer> layers = new ArrayList<>();
+        layers.add(createLayer(METADATA_LAYER_ID, "areas:100k", OskariLayer.TYPE_WFS, "https://test/wfs", null));
+        layers.add(createLayer(WMS_LAYER_ID, "areas:10k", OskariLayer.TYPE_WMS, "https://test/wms", METADATA_LAYER_ID));
+        layers.add(createLayer(SECOND_WMS_LAYER_ID, "areas:100k", OskariLayer.TYPE_WMS, "https://test/wms", null));
+        layers.add(createLayer(THIRD_WMS_LAYER_ID, "areas:1000k", OskariLayer.TYPE_WMS, "https://test/wms", WMS_LAYER_ID));
+        layers.add(createLayer(FOURTH_WMS_LAYER_ID, "areas:10000k", OskariLayer.TYPE_WMS, "https://test/wms", METADATA_LAYER_ID));
+        return layers;
+    }
+
+    private static OskariLayer createLayer(Integer wmsLayerId, String name, String layerType, String url, Integer metadataLayerId) throws JSONException {
+        OskariLayer wmsLayer = new OskariLayer();
+        wmsLayer.setId(wmsLayerId);
+        wmsLayer.setType(layerType);
+        wmsLayer.setUrl(url);
+        wmsLayer.setName(name);
+        if (metadataLayerId != null) {
+            createTimeseriesMetadata(wmsLayer, metadataLayerId);
+        }
+        return wmsLayer;
+    }
+
+    private static void createTimeseriesMetadata(OskariLayer wmsLayer, Integer metadataLayerId) throws JSONException {
+        JSONObject metadata = new JSONObject().put("layer", metadataLayerId);
+        JSONObject timeseries = new JSONObject().put("metadata", metadata);
+        wmsLayer.setOptions(new JSONObject().put("timeseries", timeseries));
+    }
+
+    @Test
+    public void testLayerUsage() throws Exception {
+        final Map<String, String> parameters = new HashMap<>();
+        parameters.put("id", METADATA_LAYER_ID.toString());
+        final ActionParameters params = createActionParams(parameters);
```

Comment:
When you add the requirement for admin user you need to do this:
```suggestion
        final ActionParameters params = createActionParams(parameters, getAdminUser());
```